### ISSUE TITLE
Rotating beyond the size of the array is identity

### ISSUE THREAD

**Issue by athas** [diku-dk/futhark#1235]
Surely that's not by design?

**Comment by philass** [comment]
It seems that this is only an issue in the Interpreter. 

When I test against the Python, C, and OpenCL backends. I get expected behaviour (not the identity).
```
Testing rotation on
\[1, 2, 3, 4, 5\]
Sanity check
testing 0
--------------------
[1i32, 2i32, 3i32, 4i32, 5i32]
^ Python
[1i32, 2i32, 3i32, 4i32, 5i32]
^ Sequential C
[1i32, 2i32, 3i32, 4i32, 5i32]
^ OpenCl
---------------------
Testing rotation of 1
---------------------
[2i32, 3i32, 4i32, 5i32, 1i32]
[2i32, 3i32, 4i32, 5i32, 1i32]
[2i32, 3i32, 4i32, 5i32, 1i32]
---------------------
Testing rotation of 6
---------------------
[2i32, 3i32, 4i32, 5i32, 1i32]
[2i32, 3i32, 4i32, 5i32, 1i32]
[2i32, 3i32, 4i32, 5i32, 1i32]
---------------------
Testing rotation of 11
---------------------
[2i32, 3i32, 4i32, 5i32, 1i32]
[2i32, 3i32, 4i32, 5i32, 1i32]
[2i32, 3i32, 4i32, 5i32, 1i32]
----------------------
Now testing negative rotations
----------------------
---------------------
Testing rotation of -1
---------------------
[5i32, 1i32, 2i32, 3i32, 4i32]
[5i32, 1i32, 2i32, 3i32, 4i32]
[5i32, 1i32, 2i32, 3i32, 4i32]
---------------------
Testing rotation of -6
---------------------
[5i32, 1i32, 2i32, 3i32, 4i32]
[5i32, 1i32, 2i32, 3i32, 4i32]
[5i32, 1i32, 2i32, 3i32, 4i32]
---------------------
Testing rotation of -11
---------------------
[5i32, 1i32, 2i32, 3i32, 4i32]
[5i32, 1i32, 2i32, 3i32, 4i32]
[5i32, 1i32, 2i32, 3i32, 4i32]
```

When tested against the Interpreter
```
[0]> rotate (1) [1, 2, 3, 4, 5]
[2i32, 3i32, 4i32, 5i32, 1i32]
[1]> rotate (6) [1, 2, 3, 4, 5]
[1i32, 2i32, 3i32, 4i32, 5i32]
[2]> rotate (11) [1, 2, 3, 4, 5]
[1i32, 2i32, 3i32, 4i32, 5i32]
[3]> rotate (-1) [1, 2, 3, 4, 5]
[5i32, 1i32, 2i32, 3i32, 4i32]
[4]> rotate (-6) [1, 2, 3, 4, 5]
[1i32, 2i32, 3i32, 4i32, 5i32]
[5]> rotate (-11) [1, 2, 3, 4, 5]
[1i32, 2i32, 3i32, 4i32, 5i32]
```
I think the backends handle this correctly and the Interpreter does not.

**Comment by athas** [comment]
Fixed by #1236.
